graph/src/blockchain/mod.rs

```diff
@@ -79,29 +64,39 @@ pub trait Blockchain: Sized + Send + Sync + 'static {
     /// Trigger filter used as input to the triggers adapter.
     type TriggerFilter: TriggerFilter<Self>;
 
-    type NodeCapabilities;
+    type NodeCapabilities: std::fmt::Display;
 
     type IngestorAdapter: IngestorAdapter<Self>;
 
     // type RuntimeAdapter: RuntimeAdapter;
     // ...WIP
 
     fn reorg_threshold() -> u32;
+
+    // ETHDEP: This assumes that capabilities are exactly the eth capabilities
+    fn node_capabilities(&self, archive: bool, traces: bool) -> Self::NodeCapabilities;
+
     fn triggers_adapter(
         &self,
-        network: &str,
-        capabilities: Self::NodeCapabilities,
-    ) -> Arc<Self::TriggersAdapter>;
+        loc: &DeploymentLocator,
+        capabilities: &Self::NodeCapabilities,
+    ) -> Result<Arc<Self::TriggersAdapter>, Error>;
 
     fn new_block_stream(
         &self,
-        current_head: BlockPtr,
+        deployment: DeploymentLocator,
+        start_blocks: Vec<BlockNumber>,
         filter: Self::TriggerFilter,
-    ) -> Result<Self::BlockStream, Error>;
+        metrics: Arc<BlockStreamMetrics>,
```

Comment:
It's a pass-through right now, and keeps the chain from having to understand what `metrics` means. Looking at `Blockchain.new_block_stream`, we could actually factor that back into a new `BlockStreamBuilder` since everything it needs from the chain is available as  public methods (mostly the chain store and the triggers adapter). But that's the other side of how generic the block stream needs to be: if we can reuse the block stream more widely, it makes sense to move the responsibility for creating one out of the chain and into a helper struct. If it's just Ethereum that uses that block stream, there's no point in splitting it out.

Reply:
Ok good point that this is more reusable.